#DPATCHLEVEL=0
# select() is to unspecific and only indicates that ANY event is available.
# XWindowEvent() blocks unless the SPECIFIC is available.
# Replace it by XNextEvent() and ignore unknown events.
--- src/libxosd/xosd.c
+++ src/libxosd/xosd.c
@@ -427,7 +427,9 @@
       continue;
     } else if (FD_ISSET(xfd, &readfds)) {
       XEvent report;
-      XWindowEvent(osd->display, osd->window, ExposureMask, &report);
+      /* There is a event, but it might not be an Exposure-event, so don't use
+       * XWindowEvent(), since that might block. */
+      XNextEvent(osd->display, &report);
       /* ignore sent by server/manual send flag */
       switch (report.type & 0x7f) {
       case Expose:
